config/connection_settings.go

```diff
@@ -38,6 +38,13 @@ const (
 	propertyKeySharedAccessKeyName = "SharedAccessKeyName"
 )
 
+// CloudConnectionInfo contains properties related to the cloud connection that may not be known at the start of the connector.
+type CloudConnectionInfo struct {
+	HubName  string
+	HostName string
+	DeviceID string
+}
+
 // SharedAccessKey contains the shared access key for generating SAS token for device authentication.
 type SharedAccessKey struct {
 	SharedAccessKeyName    string
```

Comment:
Actually there is such a term `SharedAccessKeyName`. But it is applicable for Azure IoT Hub Connection String, not IoT Hub _device_ connection string. More info: [Understand Different Connection Strings in Azure IoT Hub](https://devblogs.microsoft.com/iotdev/understand-different-connection-strings-in-azure-iot-hub/)

As the Kanto azure connector make use of Azure IoT Hub device connection string only, there is no point of having a `SharedAccessKey` struct. It can be replaced by a byte-array field.